pkg/archive/changes_unix.go

```diff
@@ -16,9 +16,9 @@ func statDifferent(oldStat *system.StatT, newStat *system.StatT) bool {
 		oldStat.UID() != newStat.UID() ||
 		oldStat.GID() != newStat.GID() ||
 		oldStat.Rdev() != newStat.Rdev() ||
-		// Don't look at size for dirs, its not a good measure of change
-		(oldStat.Mode()&unix.S_IFDIR != unix.S_IFDIR &&
-			(!sameFsTimeSpec(oldStat.Mtim(), newStat.Mtim()) || (oldStat.Size() != newStat.Size()))) {
+		(!sameFsTimeSpec(oldStat.Mtim(), newStat.Mtim()) ||
```

Comment:
Wondering if this was a deliberate choice; if no attributes are different, other than `mtime`, should the directory be considered "different"?

/cc @tonistiigi @dmcgowan

Reply:
@thaJeztah I wasn't sure, but have since dug through the history of this. It looks like it was deliberate for overlay: https://github.com/moby/moby/pull/11422, fixing https://github.com/moby/moby/issues/9874. Let me revert that bit for Linux. However, for Windows on NTFS, I'll keep the new behaviour where a modified time IS treated as a change. (I'll add a comment to that effect too.)

Reply:
@thaJeztah I've reverted the Unix bit and pushed the update. (And updated the commit description/PR description accordingly)